cmd/cloud_sql_proxy/proxy.go

```diff
@@ -113,6 +113,8 @@ func watchInstancesLoop(dir string, dst chan<- proxy.Conn, updates <-chan string
 			logging.Errorf("Error closing %q: %v", v.Addr(), err)
 		}
 	}
+
+	close(dst)
```

Comment:
I think this is needed so that proxy.Client.Run() exits? This is fairly non-obvious. Maybe this warrants a comment?

As an alternative: I *think* you could possibly move your signal handler go routine until *after* the `connSrc` value gets assigned, in which case the signal handler itself could just close *both* `updates` and `connSrc`, which might have the desired effect? But I may not be understanding this. The advantage is that all the parts needed to "please exit" would be together in one place, rather than depending on this close() deep in the code?

Reply:
The problem with closing connSrc is that it is returned as receive-only channel to the caller. Go won't let you close it. I tried :)

Reply:
also, added comment